x-pack/plugins/observability_solution/profiling/public/components/frame_information_window/apm_transactions.tsx

```diff
@@ -0,0 +1,221 @@
+    setPagination({ pageIndex: index });
+  }
+
+  const { status, data: transactionsPerServiceMap = {} } = useTimeRangeAsync(
+    ({ http }) => {
+      const pageStart = pagination.pageIndex * PAGE_SIZE;
+
+      return fetchTopNFunctionAPMTransactions({
+        http,
+        timeFrom: new Date(timeRange.start).getTime(),
+        timeTo: new Date(timeRange.end).getTime(),
+        functionName,
+        serviceNames: Object.keys(serviceNames).slice(pageStart, pageStart + PAGE_SIZE),
+      });
+    },
+    [
+      fetchTopNFunctionAPMTransactions,
+      functionName,
+      pagination.pageIndex,
+      serviceNames,
+      timeRange.end,
+      timeRange.start,
+    ]
+  );
+
+  const servicesAndTransactions: ServicesAndTransactions[] = useMemo(() => {
+    return Object.keys(serviceNames).flatMap((key) => {
+      const serviceTransactions = transactionsPerServiceMap[key];
+      return serviceTransactions?.transactions?.length
+        ? serviceTransactions.transactions.map((transaction) => ({
+            serviceName: key,
+            serviceSamples: serviceNames[key],
+            transactionName: transaction.name,
+            transactionSamples: transaction.samples,
+          }))
+        : [
+            {
+              serviceName: key,
+              serviceSamples: serviceNames[key],
+              transactionName: null,
+              transactionSamples: null,
+            },
+          ];
+    });
+  }, [serviceNames, transactionsPerServiceMap]);
+
+  const isLoadingTransactions = status !== AsyncStatus.Settled;
+
+  const columns: Array<EuiBasicTableColumn<ServicesAndTransactions>> = useMemo(
+    () => [
+      {
+        field: 'serviceName',
+        name: i18n.translate('xpack.profiling.apmTransactions.columns.serviceName', {
+          defaultMessage: 'Service Name',
+        }),
+        truncateText: true,
+        sortable: true,
+        render: (_, { serviceName }) => {
+          return (
+            <EuiLink
+              data-test-subj="profilingColumnsLink"
+              href={observabilityShared.locators.apm.serviceOverview.getRedirectUrl({
+                serviceName,
+              })}
+            >
+              {serviceName}
+            </EuiLink>
+          );
+        },
+      },
+      {
+        field: 'serviceSamples',
+        name: i18n.translate('xpack.profiling.apmTransactions.columns.serviceSamplesName', {
+          defaultMessage: 'Service Samples',
+        }),
+        width: '150px',
+        sortable: true,
+        render(_, { serviceSamples }) {
+          return asNumber(serviceSamples);
+        },
+      },
+      {
+        field: 'transactionName',
+        name: i18n.translate('xpack.profiling.apmTransactions.columns.transactionName', {
+          defaultMessage: 'Transaction Name',
+        }),
+        truncateText: true,
+        sortable: true,
+        render(_, { serviceName, transactionName }) {
```

Comment:
The API doesn't return the transaction type? it could be an _edge_ case but a service could have a 2 same transaction name but with different transaction type